server/src/main/java/io/druid/metadata/SQLMetadataSegmentManager.java

```diff
@@ -340,7 +357,9 @@ public boolean removeSegment(String ds, final String segmentID)
   @Override
   public boolean isStarted()
   {
-    return started;
+    synchronized (lock) {
```

Comment:
Perf change is unimportant? It might wait for long scheduled `poll()`

Reply:
Good point. Changed to use `ReentrantReadWriteLock`.